### PR TITLE
Implementation: ReviewsList component

### PR DESCRIPTION
**Pull request by kjniemela**
Ticket: https://trello.com/c/1SRXJWsj/50-reviews-reviewslist-component
"Build a component that displays "ReviewTile" components in a list. The list should be scrollable if necessary. Two tiles should be displayed by default, and a button at the bottom of the list should add 2 more tiles every time it is clicked until there are as many tiles as there are Reviews available."

### CODE REVIEW

**Review by brynrmrzz — COMMENTED** [review]
From what I understand it all seems good to me, I'm hesitant just because it's such a big pull request that I think another pair of eyes would be best.

File: src/components/reviews/Reviews.jsx
Comment (lbc1013) [inline src/components/reviews/Reviews.jsx:44]: I was wondering why you need 'isLoading' state in this feature? 🤔

File: src/components/reviews/Reviews.jsx
Comment (lbc1013) [inline src/components/reviews/Reviews.jsx:82]: It seems like the 'moreToLoad' is for the button that user clicks, if there's still unloaded reviews. But how can this be false when the user finish reading every reviews?
The 'reviewsPerLoad' looks like it's 2 and I wasn't sure how the newReviews.length can be changed..

**Review by lbc1013 — APPROVED** [review]
It also seems good to me except the ones that I left some comments. Actually, I wrote some comments mostly because I was wondering about how you wrote the code.

File: src/components/reviews/Reviews.jsx
Comment (kjniemela) [reply src/components/reviews/Reviews.jsx:44]: I was using it to make sure that the fetchReviews function can't be called if there's another call still waiting for a response from the server. I'm not sure it's still an issue but at one point it actually broke my widget if that happened.

File: src/components/reviews/Reviews.jsx
Comment (kjniemela) [reply src/components/reviews/Reviews.jsx:82]: reviewsPerLoad is how many new reviews show up every time you click the button. newReviews is the list of reviews that just got sent back from the server. So if the number of reviews you get back is less than you expected, you know that you must be out of reviews, and so there are no more reviews to load.

Comment (kjniemela) [reply src/components/reviews/Reviews.jsx:82]: As I was typing this comment I realized there was a bug in this implementation where the button wouldn't disappear immediately if there is an even amount of reviews... but honestly that's not a very serious bug, so I'm gonna leave it for now and make a ticket to figure out a fix tomorrow.

File: src/components/reviews/ReviewTile.jsx
Comment (anerolabs) [inline src/components/reviews/ReviewTile.jsx]: Johannes, do you need to declare state here?

File: src/components/reviews/ReviewsList.jsx
Comment (anerolabs) [inline src/components/reviews/ReviewsList.jsx]: Same here

File: tests/app.test.js
Comment (anerolabs) [inline tests/app.test.js:27]: Did removing the catch block here fix the bug you mentioned?

File: tests/reviews.test.js
Comment (anerolabs) [inline tests/reviews.test.js]: This test looks good but maybe update the name of the test?

**Review by anerolabs — CHANGES_REQUESTED** [review]

File: src/components/reviews/ReviewTile.jsx
Comment (kjniemela) [reply src/components/reviews/ReviewTile.jsx]: I do not, I've just left it in because I know I might have to declare state later. I can remove it for now if you'd like.

File: tests/app.test.js
Comment (kjniemela) [reply tests/app.test.js:27]: It did not, but it made the error less cryptic. I still don't know what's really going on with it, though.

File: tests/reviews.test.js
Comment (kjniemela) [reply tests/reviews.test.js]: Oops. On it.

**Review by anerolabs — APPROVED** [review]

**Pull request merged**
